py/torch_tensorrt/dynamo/conversion/impl/embedding.py

```diff
@@ -40,5 +44,125 @@ def embedding(
 
     # Implement embedding lookup with gather layer
     gather_layer = ctx.net.add_gather(embedding_tensor, indices_tensor, axis=0)
-    set_layer_name(gather_layer, target, name + "_gather", source_ir)
+    set_layer_name(gather_layer, target, f"{name}_gather", source_ir)
     return gather_layer.get_output(0)
+
+
+def embedding_bag(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    weight: TRTTensor,
+    indices: TRTTensor,
+    offsets: Optional[Union[torch.Tensor, np.ndarray]],
+    scale_grad_by_freq: bool,
+    mode: int,
+    sparse: bool,
+    per_sample_weights: Optional[TRTTensor],
+    include_last_offset: bool,
+) -> Tuple[TRTTensor, TRTTensor, TRTTensor, TRTTensor]:
+    """
+    This function is for calculating embedding bags.
+    In PyTorch, `offsets` is only used when input is 1D. If input is 2D of shape (B, N), it will be treated as B bags (sequences) each of fixed length N, and this will return B values aggregated in a way depending on the mode. `offsets` is ignored and required to be None in this case.
+    However, according to the schema, `offsets` is required for input with any dimensions. Accordingly, this function flattens N-D input to 1D and then to calculate embedding bags.
+    """
+    if len(indices.shape) != 1:
+        raise RuntimeError("Currently we only support 1D input.")
+        # indices = impl.shuffle.reshape(ctx, target, source_ir, f"{name}_reshape_indices", indices, (-1,))
+
+    if mode == 0:  # sum
+        reduce_op = functools.partial(
+            impl.reduce.sum, ctx=ctx, target=target, source_ir=source_ir
+        )
+        reduce_name = "sum"
+    elif mode == 1:  # mean
+        reduce_op = functools.partial(
+            impl.reduce.mean, ctx=ctx, target=target, source_ir=source_ir
+        )
+        reduce_name = "mean"
+    elif mode == 2:  # max
+        reduce_op = functools.partial(
+            impl.reduce.max,
+            ctx=ctx,
+            target=target,
+            source_ir=source_ir,
+            return_indices=False,
+        )
+        reduce_name = "max"
+    else:
+        raise RuntimeError(f"Currently we don't support mode={mode}.")
```

Comment:
OK, sounds good. So our principle is to not throw any errors from converter inside. We only convert those ops with proper args/data (by adding conditions/requirements to the capability validator). Otherwise, just let pytorch does it. Is my understanding right?

Reply:
I would say that we shouldn't throw errors for (potentially) valid inputs. For instance, if PyTorch were to allow `mode=3` in its converter in a new nightly, we shouldn't throw an error for that - we should instead just not support it.